solver/src/main/java/org/chocosolver/solver/search/strategy/selectors/variables/ConflictHistorySearch.java

```diff
@@ -93,8 +93,7 @@ public void onContradiction(ContradictionException cex) {
             // update q
             q.put(p, (1 - a) * qj + a * r.get(p));
             // update r
-            double rj = r.get(p);
-            r.put(p, 1d / (conflicts - rj + 1));
+            r.put(p, 1d / (conflicts - conflict.get(p) + 1));
```

Comment:
I think that updating r should be done after updating conflict map

Reply:
Well, I don't think so: " Once r(ci) and q(ci) are updated, #Conflicts is incremented by 1."
Moreover, if we update conflict before, r will always be equal to 1, for all propagators.

Reply:
Yes, I agree that variable `conflicts` should be incremented at the end. But `r` map is one step later from its real value with the update in this order. And, actually, I think that `r` map should be removed as saving the value of the reward is not useful because it is easy to compute and depends only on `conflicts` and `conflict.get(p)`

Reply:
You're right about removing `r` which is useless in that context